apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,294 @@
+            }
+
+          }
+
+          var lastElem = this.elements[this.elements.length - 1];
+          var clonedElem = lastElem.cloneNode(true);
+
+
+          this.elemList.insertBefore(clonedElem, this.elements[0]);
+          this.elements.splice(0, 0, clonedElem);
+          this.elements[0].classList.add('out-focus');
+          this.selectedElemIndex++;
+
+          lastElem.classList.add('removed');
+          this.elements.splice(-1, 1);
+
+          this.prevVerticalKey = 'up';
+
+
+          break;
+
+        case 'ArrowDown':
+
+          preVerticalNavigation.call(this);
+          document.getElementById('clock').style.opacity = 0;
+
+          this.elements[this.selectedElemIndex].classList.remove('selected');
+          if (this.selectedElemIndex == (this.elements.length - 3)) {
+            if (!this.elemList.dataset.scrolldown) {
+              this.elemList.style.top = parseFloat(this.elemList.style.top) - 6 + 'rem';
+              this.elemList.dataset.scrolldown = this.elemList.style.top;
+            }
+
+            this.selectedElemIndex--;
+
+            var removed = this.elements.splice(0, 1);
+
+            var cloned = removed[0].cloneNode(true);
+
+            removed[0].classList.add('removed');
+
+            this.elemList.appendChild(cloned);
+            this.elements[this.elements.length] = cloned;
+
+            this.selectedElemIndex++;
+            this.elements[this.selectedElemIndex].classList.add('selected');
+
+            this.elemList.style.top = this.elemList.dataset.scrolldown;
+          } else {
+            this.selectedElemIndex++;
+            this.elements[this.selectedElemIndex].classList.add('selected');
+
+            this.elemList.style.top = parseFloat(this.elemList.style.top) - 6 + 'rem';
+            this.elemList.dataset.scrolldown = this.elemList.style.top;
+          }
+
+          this.prevVerticalKey = 'down';
+
+          if (this.selectedElemIndex == 1) {
+            this.elemList.style.top = '0rem';
+          }
+
+          break;
+
+        case 'Accept':
+
+          window.removeEventListener('keydown', this);
+
+          if (this.elements[this.selectedElemIndex].getAttribute('id') == "moreApps") {
+            app.showMoreApps();
+          } else {
+            app.getAppByURL(this.elements[this.selectedElemIndex].dataset.manifesturl).launch();
+          }
+
+          break;
+      }
+
+    },
+
+    /**
+     * Setter to set selector
+     * @param  {string} selector [selector that the navigation can take]
+     * @return {[nothing]}          [nothing]
+     */
+    set points_selector(selector) {
+      this.selector = selector;
+      this.refresh();
+    },
+
+    /**
+     * Setter for preventDefault parameter. Default value is false.
+     * @param  {[boolean]} value [if true  preventDefault() function will be called there]
+     * @return {[nothing]}       [nothing]
+     */
+    set prevent(value) {
```

Comment:
I don't see this being used anywhere. And I can't imagine why it is needed. If you handle a key, call preventDefault on it. Otherwise don't. What am I missing?